baremetal/metal3machine_manager.go

```diff
@@ -1650,32 +1650,29 @@ func (m *MachineManager) DissociateM3Metadata(ctx context.Context) error {
 	return deleteObject(ctx, m.client, metal3DataClaim)
 }
 
-// getKubeadmControlPlaneName retrieves the KubeadmControlPlane object corresponding to the CAPI machine.
-func (m *MachineManager) getKubeadmControlPlaneName(_ context.Context) (string, error) {
-	m.Log.Info("Fetching KubeadmControlPlane name")
+// getControlPlaneName retrieves the ControlPlane object corresponding to the CAPI machine.
+func (m *MachineManager) getControlPlaneName(_ context.Context) (string, error) {
+	m.Log.Info("Fetching ControlPlane name")
 	if m.Machine == nil {
 		return "", errors.New("Could not find corresponding machine object")
 	}
 	if m.Machine.ObjectMeta.OwnerReferences == nil {
 		return "", errors.New("Machine owner reference is not populated")
 	}
 	for _, mOwnerRef := range m.Machine.ObjectMeta.OwnerReferences {
-		if mOwnerRef.Kind != "KubeadmControlPlane" {
-			continue
-		}
 		aGV, err := schema.ParseGroupVersion(mOwnerRef.APIVersion)
 		if err != nil {
 			return "", errors.New("Failed to parse the group and version")
 		}
 		if aGV.Group != controlplanev1.GroupVersion.Group {
 			continue
 		}
-		// adding prefix to KubeadmControlPlane name in order to be able to differentiate
-		// KubeadmControlPlane and MachineDeployment in case they have the same names set in the cluster.
-		m.Log.Info("Fetched KubeadmControlPlane name", "kubeadmControlPlane", "kcp-"+mOwnerRef.Name)
+		// adding prefix to ControlPlane name in order to be able to differentiate
+		// ControlPlane and MachineDeployment in case they have the same names set in the cluster.
+		m.Log.Info("Fetched ControlPlane name", "controlPlane", "kcp-"+mOwnerRef.Name)
 		return "kcp-" + mOwnerRef.Name, nil
```

Comment:
```suggestion
		return "cp-" + mOwnerRef.Name, nil
```

Reply:
Thanks - done and tests adjusted to match 👍